File: android/src/main/kotlin/com/bond/whiteboard/board/BoardAware.kt
package com.bond.whiteboard.board

import android.content.Context
import android.graphics.Color
import androidx.annotation.ColorInt
import com.bond.whiteboard.teb.MyBoardCallback
import com.tencent.teduboard.TEduBoardController

class BoardAware(private val context:Context) {


    var mBoard : TEduBoardController? =null

    var mBoardCallback: MyBoardCallback? = null

    fun destroy() {
        mBoard?.reset()
    }

    fun reset() {
        mBoard?.clear(true)
        mBoard?.reset()
        mBoard?.refresh()
    }

    fun setBackgroundColor(@ColorInt color:Int){
        mBoard?.backgroundColor= TEduBoardController.TEduBoardColor(color)
        mBoard?.refresh()
    }
}

File: ios/Classes/board/BoardAware.swift
//
// Created by vicky Leu on 2021/5/19.
//

import Foundation
public class BoardAware{

    var mBoard :TEduBoardController?
    var mBoardCallback:MyBoardCallback?
    func destroy() {
        mBoard?.reset()
    }
    func reset(){
        mBoard?.clearDraws()
        mBoard?.reset()
        mBoard?.refresh()
    }
    
    func setBackgroundColor(_ color:UIColor){
        mBoard?.setBackgroundColor(color)
        mBoard?.refresh()
    }
}
